### DISCUSSION TITLE
ed25519-sk doesn't work, but ecdsa-sk does, despite my security key being FIDO

### DISCUSSION THREAD

**Discussion by ghost** [Yubico/libfido2#576] [Q&A] [answered]
Hello there, I've been trying to build an SSH key with my work-issued security key recently, and I found out that while issuing the command `ssh-keygen` with the `t` parameter set to `ed25519-sk` as the [arch wiki](https://wiki.archlinux.org/title/SSH_keys#FIDO/U2F) suggests. For some reason, when I run the command, I get the following almost immediately:
[output](https://pastebin.com/hmRCtbpN)
I tried searching up what the issue might be, found an issue in this repo somewhat related to my issue, but I couldn't find anything really helpful. Although, when I tried using `ecdsa-sk` instead of ed25519, I got it to work just fine.
[output](https://pastebin.com/N0G7Fz0K)
If anyone knows what might be going or, what I could possibly try to do to see if it might help, I'm all ears
Some side notes: I don't fully know the capabilities of this dongle, and I've tried to search something equivalent to yubikey-manager for other kinds of security keys with no luck. If someone knows about anything similar so I can see what it is really capable of, I'd appreciate it if you could say it.
And also, I am not sure if this is the correct place to ask this question, so please tell me so and I'll close the discussion

**Comment by martelletto** [comment, marked as answer]
The security key is likely a U2F (FIDO) key. U2F keys only support ECDSA. You can use `fido2-token -I /dev/hidrawX` to obtain information on the key's capabilities. fido2-token is part of the libfido2 package in Arch Linux.
